TarponSimulator2017/Core/World.cs

```diff
@@ -13,21 +13,57 @@ public class World : GameObject
 
 		public List<IUpdatable> toUpdate { get; private set; }
 
-		public World ()
+		private static World UniqueInstance;
+
+		public static World Instance {
+			get {
+				if (UniqueInstance == null) {
+					UniqueInstance = new World ();
+				}
+				return UniqueInstance;
+			}
+		}
+
+
+		public void InitWorld ()
 		{
-			playerBoat = new Boat (200, 300);
+			playerBoat = new Boat (950, 540);
 			playerBoat.FrameOfReference = this;
-			FirstFish = new Fish (400, 400);
-			FirstFish.FrameOfReference = this;
 			toUpdate = new List<IUpdatable> ();
 			toUpdate.Add (playerBoat);
-			toUpdate.Add (FirstFish);
+			//Handle Fish Generation
+			IList<Fish> ListOfFishes = FishFactory.Instance.InitFish (1);
+			this.AddAListOfFish (ListOfFishes);
+		}
+
+		public void AddAfFish (Fish f)
```

Comment:
Wow, I really don't like these list of fishes, sorry :(
I don't totally understand what they are, but still they are everywhere.
You know, the problem is not when there is one, but many...

Couldn't we have only a list of fishes in the world?